homework/src/index.js

```diff
@@ -1,47 +1,91 @@
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
-    });
-  }
-
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-
-  window.onload = () => main(HYF_REPOS_URL);
-}
+    function fetchJSON(url, cb) {
+        const xhr = new XMLHttpRequest();
+        xhr.open('GET', url);
+        xhr.responseType = 'json';
+        xhr.onload = () => {
+            if (xhr.status < 400) {
+                cb(null, xhr.response);
+            } else {
+                cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+            }
+        };
+        xhr.onerror = () => cb(new Error('Network request failed'));
+        xhr.send();
+    }
+
+    function createAndAppend(name, parent, options = {}) {
+        const elem = document.createElement(name);
+        parent.appendChild(elem);
+        Object.keys(options).forEach((key) => {
+            const value = options[key];
+            if (key === 'text') {
+                elem.innerHTML = value;
+            } else {
+                elem.setAttribute(key, value);
+            }
+        });
+        return elem;
+    }
+
+    function main(url) {
+
+
+        const root = document.getElementById('root');
+
+        const div = createAndAppend('div', root, { class: 'container' });
+        const header = createAndAppend('h3', div, { text: "HYF Repositories  ", class: 'header' });
+        const select = createAndAppend('select', header, { class: "select" });
+        const table = createAndAppend('table', root, { class: 'table' });
+        const tbody = createAndAppend('tbody', table);
+        const tr = createAndAppend('tr', tbody);
+        createAndAppend('td', tbody, { text: "Repository: ", class: 'label' });
+        const repoTitle = createAndAppend('a', tbody);
+        const descriptionTitle = createAndAppend('tr', tbody);
+        const forksTitle = createAndAppend('tr', tbody, { class: 'label' });
+
+
+        const updateTitle = createAndAppend('tr', tbody, { class: 'label' });
+        const zz = createAndAppend('td', tbody);
+        //calling function inside 'main'
+        fetchJSON(url, (err, data) => {
+
+            data.sort(function(a, b) {
+                return a.name.localeCompare(b.name);
+
+            });
+
+            if (err) {
+                createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+            } else {
+                data.forEach((item, index) => {
+                    const option = createAndAppend('option', select, { text: item.name, value: index });
+                });
+
+
+                select.addEventListener('change', (event) => {
+
+                    const index = event.target.value;
+
+                    repoTitle.innerText = select[index].textContent;
```

Comment:
`repoTitle` and `forksTitle` refer to `tr` elements. These elements should only contain `td` elements (or `th` elements in case of a header). Inside the `td` element you can define the actual content, such as text nodes and for instance `a` elements.